spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -512,7 +511,7 @@ public Object doInSession(Session<F> session) throws IOException {
 
 	private Object doLs(Message<?> requestMessage) {
 		String dir = this.fileNameProcessor.processMessage(requestMessage);
-		if (!dir.endsWith(this.remoteFileTemplate.getRemoteFileSeparator())) {
+		if (dir != null && !dir.endsWith(this.remoteFileTemplate.getRemoteFileSeparator())) {
 			dir += this.remoteFileTemplate.getRemoteFileSeparator();
 		}
 		final String fullDir = dir;
```

Comment:
Even though it won't hurt we should probably not `.setHeader` below if it's `null`.